arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1405,11 +1417,27 @@ Result RocksDBCollection::insertDocument(transaction::Methods* trx,
     return res.reset(TRI_ERROR_DEBUG);
   }
 
-  rocksdb::Status s = mthds->PutUntracked(
-      RocksDBColumnFamilyManager::get(
-          RocksDBColumnFamilyManager::Family::Documents),
-      key.ref(),
-      rocksdb::Slice(doc.startAs<char>(), static_cast<size_t>(doc.byteSize())));
+  rocksdb::Status s;
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+  // use Put() instead of PutUntracked()
+  if (options.isRestore) {
+    s = mthds->Put(RocksDBColumnFamilyManager::get(
+                       RocksDBColumnFamilyManager::Family::Documents),
+                   key.ref(),
+                   rocksdb::Slice(doc.startAs<char>(),
+                                  static_cast<size_t>(doc.byteSize())),
+                   /*assume_tracked*/ false);
```

Comment:
`Put` will only check for conflicts with concurrent operations, but would still silently overwrite any already existing key. We should instead use a `GetForUpdate` to check for an existing key and lock it. Then we can even stick with `PutUntracked`.

Reply:
Addressed.